cmd/apiProviderDownload_test.go

```diff
@@ -0,0 +1,88 @@
+package cmd
+
+import (
+	"testing"
+
+	"github.com/SAP/jenkins-library/pkg/mock"
+	"github.com/stretchr/testify/assert"
+)
+
+type apiProviderDownloadTestUtilsBundle struct {
+	*mock.ExecMockRunner
+	*mock.FilesMock
+}
+
+func apiProviderDownloadMockUtilsBundle() *apiProviderDownloadTestUtilsBundle {
+	utilsBundle := apiProviderDownloadTestUtilsBundle{
+		ExecMockRunner: &mock.ExecMockRunner{},
+		FilesMock:      &mock.FilesMock{},
+	}
+	return &utilsBundle
+}
+
+//Successful API Provider download cases
+func TestApiProviderDownloadSuccess(t *testing.T) {
+	t.Parallel()
+	t.Run("Successful Download of API Provider", func(t *testing.T) {
+		apiServiceKey := `{
+			"oauth": {
+				"url": "https://demo",
+				"clientid": "demouser",
+				"clientsecret": "******",
+				"tokenurl": "https://demo/oauth/token"
+			}
+		}`
+
+		config := apiProviderDownloadOptions{
+			APIServiceKey:   apiServiceKey,
+			APIProviderName: "provider1",
+			DownloadPath:    "APIProvider.json",
+		}
+		// test
+		httpClient := httpMockCpis{CPIFunction: "APIProviderDownload", ResponseBody: ``, TestType: "Positive"}
+		utilsMock := apiProviderDownloadMockUtilsBundle()
+		err := runApiProviderDownload(&config, nil, &httpClient, utilsMock)
+
+		if assert.NoError(t, err) {
+			t.Run("Assert file download", func(t *testing.T) {
+				assert.True(t, utilsMock.HasWrittenFile(config.DownloadPath))
```

Comment:
Please also check the content is as expected. The idea of this test is not to check whether FileWrite works as expected, but to make sure the expected content is written to the expected location.

Reply:
Added one more test to assert the file content as string. Thanks.